src/main/java/org/plp/implementations/plpisa/PlpFile.java

```diff
@@ -64,7 +72,7 @@ public String getFilePath() {
      */
     @Override
     public boolean addInstructionToFile(String instruction) {
-        return false;
+        return fileContent.add(instruction);
```

Comment:
instruction = "" and instruction = null;

I think the previous one can be added to the fileContent but later can not.

And we need to test the exception on the **instruction = null**

Reply:
Going to use [Lombok](https://objectcomputing.com/resources/publications/sett/january-2010-reducing-boilerplate-code-with-project-lombok) here with NonNull which will automatically going to throw a NullPointerException.